src/dash-table/derived/edges/type.ts

```diff
@@ -0,0 +1,197 @@
+import * as R from 'ramda';
+import { CSSProperties } from 'react';
+
+import { OptionalMap, OptionalProp, PropOf } from 'core/type';
+import py2jsCssProperties from '../style/py2jsCssProperties';
+
+export type Edge = any;
+
+type BorderProp =
+    PropOf<CSSProperties, 'borderBottom'> |
+    PropOf<CSSProperties, 'borderLeft'> |
+    PropOf<CSSProperties, 'borderRight'> |
+    PropOf<CSSProperties, 'borderTop'>;
+
+export type BorderStyle =
+    OptionalMap<CSSProperties, 'borderBottom', [OptionalProp<CSSProperties, 'borderBottom'>, number]> &
+    OptionalMap<CSSProperties, 'borderLeft', [OptionalProp<CSSProperties, 'borderLeft'>, number]> &
+    OptionalMap<CSSProperties, 'borderRight', [OptionalProp<CSSProperties, 'borderRight'>, number]> &
+    OptionalMap<CSSProperties, 'borderTop', [OptionalProp<CSSProperties, 'borderTop'>, number]>;
+
+export const BORDER_PROPERTIES: BorderProp[] = [
+    'borderBottom',
+    'borderLeft',
+    'borderRight',
+    'borderTop'
+];
+
+export const BORDER_PROPERTIES_AND_FRAGMENTS: string[] = R.uniq(
+    R.filter(
+        p => p.indexOf('border') === 0,
+        Array.from(py2jsCssProperties.values())
+    )
+);
```

Comment:
For use later, will omit all `border**` rules from the styles applied to the table cells